src/algorithms/simulation/circuit_to_truthtable.cpp

```diff
@@ -0,0 +1,25 @@
+#include "algorithms/simulation/circuit_to_truthtable.hpp"
+
+namespace syrec {
+
+    auto buildTruthTable(const qc::QuantumComputation& qc, TruthTable& tt) -> void {
+        const auto nBits = qc.getNqubits();
+        auto       dd    = std::make_unique<dd::Package<>>(nBits);
+
+        const auto totalInputs = 1U << nBits;
+
+        std::uint64_t n = 0U;
+
+        while (n < totalInputs) {
+            const auto inCube = TruthTable::Cube::fromInteger(n, nBits);
+
+            auto const inEdge    = dd->makeBasisState(nBits, inCube.toBoolVec());
+            const auto out       = dd::simulate(&qc, inEdge, dd, 1);
+            const auto outString = out.begin()->first;
+
```

Comment:
This is unnecessarily limited to <64 qubits. Just straight out compute `std::vector<bool>` for the simulation. Or, if that should be too cumbersome, construct a bitset with as many bits as maximally supported by the DDPackage.

On the other hand, you'll never build a 64Bit truth table. In that case at least add an assertion here that makes sure no one calls this function with more than 64 qubits.